index.bs

```diff
@@ -1641,6 +1645,28 @@ session.SystemProxyConfiguration = (
 
 </pre>
 
+#### The session.UserPromptHandler Type #### {#type-session-UserPromptHandler}
+
+[=Remote end definition=] and [=local end definition=]
+
+<pre class="cddl remote-cddl local-cddl">
+session.UserPromptHandler = session.UserPromptHandlerType /
```

Comment:
We should also keep support for the string type capability given that `accept and notify` and `dismiss and notify` are also valid options, and those don't match with the `UserPromptHandlerConfiguration` type.

Reply:
The `and notify` variants are just for the `Unexpected User Prompt Open` error and so don't have any effect on BiDi-only sessions. You can still subscribe to the events and if there's a prompt handler installed you'll get events for the prompt opening and being dismissed. Therefore I don't think this makes sense.

Reply:
So what should happen if no event handler is set for a BiDi-only session, which means that prompts are not handled in the case of the `ignore` setting. Does it mean we always raise an error? But when? None of the commands for BiDi check if a prompt is still open and potentially would fail with some random error depending on what the command is trying to accomplish.

Reply:
It means you get events and have to handle the prompts yourself using `browsingContext.handleUserPrompt`.

I don't think other commands should error, but they might be blocked on the prompt if they need to interact with the js event loop. Some commands don't (e.g. those that can run entirely in the parent process). It's probably technically specified how this works, but I kind of doubt the specifications match browsers in practice. 

We might want to tighten this up in the longer term, but I don't know what we could say in the short term. Unless we have some evidence that say CDP takes a specific approach here that we'd like to copy.

Reply:
Ok, as it sounds like it worth a follow-up issue then, also given that other prompt types will trigger the exact same situation and it's not just related to `beforeunload` prompts.

Reply:
Just one more question in regards of multiple session... When we have a classic session with `dismiss and notify`, and a BiDi client wants to create another session will that actually work with capability matching when it specifies `dismiss` as handler? The `dismiss and notify` will not work as you said above.

Reply:
https://w3c.github.io/webdriver/#dfn-check-user-prompt-handler-matches says that we fail to create the session in that case, but I think you're correct that we could relax that to not care about the "notify" part, since that can only be set by HTTP sessions, doesn't affect, BiDi, and you can only have a single HTTP session.